File: app/src/main/java/com/example/yeshasprabhakar/todo/TodoRepository.java
package com.example.yeshasprabhakar.todo;

import android.app.Application;
import androidx.lifecycle.LiveData;

import java.util.List;

public class TodoRepository {

    private TodoDao mTodoDao;
    private LiveData<List<TodoItem>> mAllTodos;

    public TodoRepository(Application application) {
        AppDatabase db = AppDatabase.getDatabase(application);
        mTodoDao = db.todoDao();
        mAllTodos = mTodoDao.getAllTodos();
    }

    public LiveData<List<TodoItem>> getAllTodos() {
        return mAllTodos;
    }

    public void insert(TodoItem todoItem) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            mTodoDao.insert(todoItem);
        });
    }

    public void deleteByNameDateTime(String name, String date, String time) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            mTodoDao.deleteByNameDateTime(name, date, time);
        });
    }
}

File: app/src/main/java/com/example/yeshasprabhakar/todo/TodoRepository.kt
package com.example.yeshasprabhakar.todo

import android.app.Application
import androidx.lifecycle.LiveData

class TodoRepository(application: Application) {

    private val todoDao: TodoDao // Renamed for Kotlin style (m-prefix removed)
    val allTodos: LiveData<List<TodoItem>> // Renamed and made public val

    init {
        val db = AppDatabase.getDatabase(application)
        todoDao = db.todoDao()
        allTodos = todoDao.getAllTodos()
    }

    fun insert(todoItem: TodoItem) {
        AppDatabase.databaseWriteExecutor.execute {
            todoDao.insert(todoItem)
        }
    }

    fun deleteByNameDateTime(name: String, date: String, time: String) {
        AppDatabase.databaseWriteExecutor.execute {
            todoDao.deleteByNameDateTime(name, date, time)
        }
    }
}
